README.rst

```diff
@@ -306,6 +306,29 @@ Any valid ``pip`` flags or arguments may be passed on with ``pip-compile``'s
 Configuration
 -------------
 
+You can define project-level defaults for ``pip-compile`` and ``pip-sync`` by
+writing them to a configuration file in the same directory as your requirements
+input file. By default, both ``pip-compile`` and ``pip-sync`` will look first
+for a ``.pip-tools.toml`` file and then in your ``pyproject.toml``. You can
+also specify an alternate TOML configuration file with the ``--config`` option.
+
+For example, to by default generate ``pip`` hashes in the resulting
+requirements file output, you can specify in a configuration file
+
+.. code-block:: toml
+
+    # In a .pip-tools.toml file
+    [pip-tools]
```

Comment:
I'm unsure about this. The section name could probably be the same as for `pyproject.toml` which would make it easier to migrate between the files... OTOH, `tool.` might be excessive here.
```suggestion
    [tool.pip-tools]
```
This is not a request to make the change right away but rather an idea to be discussed.

Reply:
Certainly - happy to have that discussion. I was comparing with other projects and trying for consistency. You can see, for example, that in this project's `pyproject.toml` that mypy, setuptools, setuptools_scm, and pytest all have the `tool` prefix - whereas flake8 and bandit's bespoke files just use the name of the tool as the section of the file.

Reply:
Yeah, I mostly just wanted to bring attention to this aspect. You wanted to ensure the implementation remains simple, and I saw a possibility to make that happen.

On a more general note, I understand that the tools listed come with a lot of legacy and their tool-specific configs don't use TOML at all.

I have a precedent for you, though. Towncrier, for example, expects `[tool.towncrier]` in both of its supported config file locations — `towncrier.toml` and `pyproject.toml`. And this makes sense — what if an end-user passes a config file path with an arbitrary name of `my-cool-custom-config.toml` — would we expect the section to be `[pip-tools]` or `[tool.pip-tools]`?

Reply:
I notice that for [ruff](https://beta.ruff.rs/docs/configuration/#using-rufftoml), when using a `ruff.toml`/`.ruff.toml`, the settings can either be under `[tool.ruff]`, or without a top level heading at all.

Reply:
My original motivation was to help simplify the patch and the code base in general. Adding a conditional would keep some of the complexity. Do we really want that in order to save people one line? I'm conflicted here. Though, it's not as important to me so I won't be insisting of a specific resolution of this.

Reply:
I think my recommendation would be to maintain consistency with the rest of the ecosystem insofar as we can ascertain it. I would accept an incremental degree of code complexity to achieve a better developer experience with minimal cognitive load.

Surveying popular tools in the Python ecosystem, it seems that `[tool.pip-tools]` is the convention for TOML files and `[pip-tools]` would be the convention for INI-style files. I'll make it consistent, since we're using TOML everywhere.

Reply:
Agreed. Marking this thread as resolved.